src/Request/BasicResolver.php

```diff
@@ -58,10 +58,17 @@ protected static function getRequestHeaders(array $server)
 
         $headers = [];
 
-        foreach ($server as $name => $value) {
-            if (substr($name, 0, 5) == 'HTTP_') {
-                $headers[str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))))] = $value;
+        if (!function_exists('getallheaders')) {
```

Comment:
Small nitpick — if there's an else then I usually like the "positive" case first, i.e.

```php
if (function_exists('getallheaders')) {
    $headers = getallheaders();
} else {
    // ...
}
```

We could even avoid the else entirely:

```php
if (function_exists('getallheaders')) {
    $headers = getallheaders();

    return $headers;
}

$headers = [];

foreach ($server as $name => $value) {
    // ...
}
```